pkg/cloudprovider/aws/instancetypes.go

```diff
@@ -89,13 +89,17 @@ func (p *InstanceTypeProvider) newInstanceType(ctx context.Context, info *ec2.In
 		provider:         provider,
 		offerings:        offerings,
 	}
+	opts := injection.GetOptions(ctx)
 	// Precompute to minimize memory/compute overhead
-	instanceType.resources = instanceType.computeResources(injection.GetOptions(ctx).AWSEnablePodENI)
+	instanceType.resources = instanceType.computeResources(opts.AWSEnablePodENI)
 	instanceType.overhead = instanceType.computeOverhead()
 	instanceType.requirements = instanceType.computeRequirements()
-	if !injection.GetOptions(ctx).AWSENILimitedPodDensity {
+	if !opts.AWSENILimitedPodDensity {
 		instanceType.maxPods = ptr.Int32(110)
 	}
+	if opts.AWSVolumeAttachLimit >= 0 {
+		instanceType.maxVolumes = ptr.Int32(int32(opts.AWSVolumeAttachLimit))
```

Comment:
This seems wrong. IIUC, the EBS CSI driver estimates this if `=-1`. Can we try to approximate?